tt_metal/impl/dispatch/command_queue.cpp

```diff
@@ -393,21 +352,82 @@ const DeviceCommand EnqueueReadBufferCommand::assemble_device_command(uint32_t d
     command.set_consumer_cb_num_pages(consumer_cb_num_pages);
     command.set_num_pages(num_pages);
 
-    if (is_sharded(this->buffer.buffer_layout())) {
-        uint32_t num_cores = this->buffer.num_cores();
-        auto core_bank_indices = this->buffer.core_bank_indices();
-        command.set_sharded_buffer_num_cores(num_cores);
+    return command;
+}
+
+const DeviceCommand EnqueueReadBufferCommand::assemble_device_command_sharded(uint32_t dst_address) {
```

Comment:
would it make sense to create a new class `EnqueueReadShardedBufferCommand` for shard specific code ?

Reply:
I agree. I think EnqueueReadBuffer/EnqueueWriteBuffer can behind the scenes call the relevant API.